openstudiocore/src/openstudio_lib/GridItem.cpp

```diff
@@ -1185,13 +1187,47 @@ HorizontalBranchGroupItem::HorizontalBranchGroupItem( model::Splitter & splitter
         auto comp1 = it1->optionalCast<model::HVACComponent>();
         OS_ASSERT(comp1);
         auto branchComponents = loop->components(comp1.get(),mixer);
-        branchComponents.pop_back();
 
-        if( isSupplySide ) {
-          allBranchComponents.push_back(branchComponents);
+        // Can't pop_back if it's empty to begin with...
+        if (branchComponents.empty()) {
+          std::stringstream ss;
+          ss << "Found orphaned component while drawing loop for " << comp1.get().briefDescription() << ".";
+
+          if (comp1->isRemovable()) {
+            ss << " Removing it.";
+            comp1->remove();
+          } else {
+
+            ss << " But this component is not removable. You should use the Ruby bindings to disconnect then remove it";
```

Comment:
That `remove` part isn't really needed.
And again, this type of stuff shouldn't happen to begin with, in case of #3116 it's because the user modified the file manually in a text editor. I still think we should at least catch that pop_back error and issue a message to the user. To me this is very similar to the sketchup plugin catching surface errors (mismatched references, wrong orientation, etc).